maint/local_python/binding_f08.py

```diff
@@ -1575,16 +1575,19 @@ def get_int_type(fortran_type):
         elif "MPI_COUNT_KIND" in fortran_type:
             return "count"
         else:
-            raise Exception("Unrecognized POLY type")
+            raise Exception("Unrecognized POLY int type: %s" % fortran_type)
 
     small_map = G.MAPS['SMALL_F08_KIND_MAP']
     large_map = G.MAPS['BIG_F08_KIND_MAP']
     for kind in small_map:
-        if small_map[kind].startswith('POLY'):
+        if kind == 'POLYFUNCTION':
+            # TODO: potentially tricky. Might be easier to treat individual function case by case.
+            G.real_poly_kinds[kind] = 1
+        elif kind.startswith('POLY'):
```

Comment:
what is the difference between `kind == POLYFUNCTION` and kind.startswith('POLY')`?

Reply:
For latter condition includes the former. We need single `POLYFUNCTION` out because it is not differentiated in the mappings between `SMALL` and `BIG` but they still need be generated for both since the actual function prototype is different between `SMALL` and `BIG`

Reply:
OK, I think I understand what you're saying. We need both prototypes because the integer kinds are different, even though the size of those kinds are the same. Small and big mappings only apply to functions where the integer kinds are of different sizes.